server/src/main/java/org/elasticsearch/index/fielddata/BooleanScriptFieldData.java

```diff
@@ -78,10 +89,12 @@ protected boolean sortRequiresCustomComparator() {
 
     public static class BooleanScriptLeafFieldData extends LeafLongFieldData {
         private final BooleanScriptDocValues booleanScriptDocValues;
+        protected final ToScriptField<SortedNumericDocValues> toScriptField;
 
-        BooleanScriptLeafFieldData(BooleanScriptDocValues booleanScriptDocValues) {
-            super(0, NumericType.BOOLEAN);
+        BooleanScriptLeafFieldData(BooleanScriptDocValues booleanScriptDocValues, ToScriptField<SortedNumericDocValues> toScriptField) {
```

Comment:
Is this always `BooleanDocValuesField::new`? In which case we can probably get away without having a field here?

Reply:
Good catch! This is a mistake on my part and is now fixed. I went through the rest of the types that extend NumericIndexFieldData to ensure I didn't make this same mistake elsewhere.

The parent mapped type is already passing in `toScriptField` from the IndexFieldData.Builder, and I just didn't use it. For consistency, I would like to take all of `toScriptField` to come from the parent mapped type for now even when there may only be a singular field type. I think that'll make the general code easier to follow/maintain and allows mapped types to re-use the IndexFieldData from plugins in the way unsigned long works.